### ISSUE TITLE
"nmstate apply" fails when a new bonding option is added to the kernel

### ISSUE THREAD

**Issue by VenkatramananV** [nmstate/nmstate#2464]
## Describe the bug

In our Oracle Linux kernel if we add a new bonding option,
we are not able to configure the bond interface anymore
even though the new bonding option is not used by the 
bonding interface configuration. 

## Nmstate version
[root@localhost ~]# nmstatectl -V
nmstatectl 2.2.7

## NetworkManager version
[root@localhost ~]# NetworkManager -V
1.40.0-1.0.2.el9

## Platform
Oracle Lunux 9.1

[root@localhost ~]# cat /etc/oracle-release
Oracle Linux Server release 9.1

## How reproducible
- [e.g. 100%, random]

100%

## Steps to Reproduce

# nmstatectl apply create-bond.xml

Pasted create-bond.xml below:

```yml
---
interfaces:
- name: bond0
  type: bond
  state: up
  ipv4:
    enabled: true
    address:
    - ip: 10.211.14.70
      prefix-length: 22
    dhcp: false
  link-aggregation:
    mode: active-backup
    port:
    - ens3
    - ens4
- name: ens3
  type: ethernet
  state: up
- name: ens4
  type: ethernet
  state: up

dns-resolver:
  config:
    search:
    - us.oracle.com
    server:
    - 206.223.27.1
```

## Nmstate output
- Expected output:

```
[2023-11-20T12:39:56Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T12:39:56Z INFO  nmstate::query_apply::net_state] Created checkpoint /org/freedesktop/NetworkManager/Checkpoint/2
[2023-11-20T12:39:56Z INFO  nmstate::ifaces::inter_ifaces] Ignoring interface lo type loopback
[2023-11-20T12:39:56Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("813d5277-af68-4b18-a337-ec013f79ba4c"), ID Some("bond0"), type Some("bond") name Some("bond0")
[2023-11-20T12:39:56Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("ea40f610-16f1-4a70-b630-470d68ad0802"), ID Some("ens3"), type Some("802-3-ethernet") name Some("ens3")
[2023-11-20T12:39:56Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("fc29a19c-b18f-42e7-a8b6-7f54e15b3e05"), ID Some("ens4"), type Some("802-3-ethernet") name Some("ens4")
[2023-11-20T12:39:56Z INFO  nmstate::nm::query_apply::profile] Activating connection 813d5277-af68-4b18-a337-ec013f79ba4c: bond0/bond
[2023-11-20T12:39:56Z INFO  nmstate::nm::query_apply::profile] Reapplying connection ea40f610-16f1-4a70-b630-470d68ad0802: ens3/802-3-ethernet
[2023-11-20T12:39:56Z INFO  nmstate::nm::query_apply::profile] Reapplying connection fc29a19c-b18f-42e7-a8b6-7f54e15b3e05: ens4/802-3-ethernet
[2023-11-20T12:39:57Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T12:39:57Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.link-aggregation.port desire '["ens3","ens4"]', current '["ens3"]'
[2023-11-20T12:39:58Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T12:39:58Z INFO  nmstate::query_apply::net_state] Destroyed checkpoint /org/freedesktop/NetworkManager/Checkpoint/2
dns-resolver:
  config:
    server:
    - 206.223.27.1
    search:
    - us.oracle.com
route-rules: {}
routes: {}
interfaces:
- name: bond0
  type: bond
  state: up
  ipv4:
    enabled: true
    dhcp: false
    address:
    - ip: 10.211.14.70
      prefix-length: 22
  link-aggregation:
    mode: active-backup
    port:
    - ens3
    - ens4
- name: ens3
  type: ethernet
  state: up
- name: ens4
  type: ethernet
  state: up
ovs-db: {}
```

- Current output:

```
[2023-11-20T13:11:05Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:05Z INFO  nmstate::query_apply::net_state] Created checkpoint /org/freedesktop/NetworkManager/Checkpoint/2
[2023-11-20T13:11:05Z INFO  nmstate::ifaces::inter_ifaces] Ignoring interface lo type loopback
[2023-11-20T13:11:05Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("813d5277-af68-4b18-a337-ec013f79ba4c"), ID Some("bond0"), type Some("bond") name Some("bond0")
[2023-11-20T13:11:05Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("ea40f610-16f1-4a70-b630-470d68ad0802"), ID Some("ens3"), type Some("802-3-ethernet") name Some("ens3")
[2023-11-20T13:11:05Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("fc29a19c-b18f-42e7-a8b6-7f54e15b3e05"), ID Some("ens4"), type Some("802-3-ethernet") name Some("ens4")
[2023-11-20T13:11:05Z INFO  nmstate::nm::query_apply::profile] Activating connection 813d5277-af68-4b18-a337-ec013f79ba4c: bond0/bond
[2023-11-20T13:11:05Z INFO  nmstate::nm::query_apply::profile] Reapplying connection ea40f610-16f1-4a70-b630-470d68ad0802: ens3/802-3-ethernet
[2023-11-20T13:11:05Z INFO  nmstate::nm::query_apply::profile] Reapplying connection fc29a19c-b18f-42e7-a8b6-7f54e15b3e05: ens4/802-3-ethernet
[2023-11-20T13:11:05Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:05Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:06Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:06Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:07Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:07Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:08Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:08Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:09Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:09Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:10Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("813d5277-af68-4b18-a337-ec013f79ba4c"), ID Some("bond0"), type Some("bond") name Some("bond0")
[2023-11-20T13:11:10Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("ea40f610-16f1-4a70-b630-470d68ad0802"), ID Some("ens3"), type Some("802-3-ethernet") name Some("ens3")
[2023-11-20T13:11:10Z INFO  nmstate::nm::query_apply::profile] Modifying connection UUID Some("fc29a19c-b18f-42e7-a8b6-7f54e15b3e05"), ID Some("ens4"), type Some("802-3-ethernet") name Some("ens4")
[2023-11-20T13:11:10Z INFO  nmstate::nm::query_apply::profile] Activating connection 813d5277-af68-4b18-a337-ec013f79ba4c: bond0/bond
[2023-11-20T13:11:10Z INFO  nmstate::nm::query_apply::profile] Reapplying connection ea40f610-16f1-4a70-b630-470d68ad0802: ens3/802-3-ethernet
[2023-11-20T13:11:10Z INFO  nmstate::nm::query_apply::profile] Reapplying connection fc29a19c-b18f-42e7-a8b6-7f54e15b3e05: ens4/802-3-ethernet
[2023-11-20T13:11:10Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:10Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:11Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:11Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:12Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:12Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:13Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:13Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:13Z INFO  nmstate::query_apply::net_state] Retrying on: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
[2023-11-20T13:11:14Z INFO  nmstate::nm::show] Got unsupported interface type generic: lo, ignoring
[2023-11-20T13:11:14Z INFO  nmstate::query_apply::net_state] Rollbacked to checkpoint /org/freedesktop/NetworkManager/Checkpoint/2
NmstateError: VerificationError: Verification failure: bond0.interface.ipv4.address desire '[{"ip":"10.211.14.70","prefix-length":22}]', current 'null'
```

### Additional context

1) If we run it with --no-verify it works.

2a)  IFLA_BOND_ARP_NEWOPTION is the new option added below in the kernel.

[include/uapi/linux/if_link.h]

```
/* Bonding section */

enum {
...
        IFLA_BOND_PEER_NOTIF_DELAY,
        IFLA_BOND_AD_LACP_ACTIVE,
        IFLA_BOND_MISSED_MAX,
        IFLA_BOND_NS_IP6_TARGET,
        IFLA_BOND_ARP_NEWOPTION,
        __IFLA_BOND_MAX,
}
```
 
2b) If the bonding driver doesn't report the new bonding option to netlink (for example to RTM_GETLINK),
      then nmstate doesn't complain and it is able to configure the bond interface fine.

i.e. if we simply comment out the below code in the bonding driver, nmstate works fine.

drivers/net/bonding/bond_netlink.c:

```
static int bond_fill_info(struct sk_buff *skb,
                          const struct net_device *bond_dev)
{
...
#if 0
        if (nla_put_u8(skb, IFLA_BOND_ARP_NEWOPTION,
                       bond->params.arp_allslaves))
                goto nla_put_failure;
#endif

        return 0;

nla_put_failure:
        return -EMSGSIZE;
}
```

2c) Output of "npc iface bond0" with the "2b" experiment.

```
[2023-11-20T12:42:19Z WARN  nispor::netlink::bond] unknown nla kind 9 value: [0, 0, 0, 0]
[2023-11-20T12:42:19Z WARN  nispor::netlink::bond] unknown nla kind 9 value: [0, 0, 0, 0]
[2023-11-20T12:42:19Z WARN  nispor::ifaces::bond] Unsupported InfoBond: ActiveSlave(4)
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolRingAttr TxPush(false)
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolRingAttr Other(DefaultNla { kind: 14, value: [0] })
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolRingAttr TxPush(false)
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolRingAttr Other(DefaultNla { kind: 14, value: [0] })
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolRingAttr TxPush(false)
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolRingAttr Other(DefaultNla { kind: 14, value: [0] })
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolRingAttr TxPush(false)
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolRingAttr Other(DefaultNla { kind: 14, value: [0] })
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolLinkModeAttr Other(DefaultNla { kind: 10, value: [0] })
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolLinkModeAttr Other(DefaultNla { kind: 10, value: [0] })
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolLinkModeAttr Other(DefaultNla { kind: 10, value: [0] })
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolLinkModeAttr Other(DefaultNla { kind: 10, value: [0] })
[2023-11-20T12:42:19Z WARN  nispor::ifaces::ethtool] WARN: Unsupported EthtoolLinkModeAttr Other(DefaultNla { kind: 10, value: [0] })
- name: bond0
  iface_type: bond
  state: up
  mtu: 1500
  min_mtu: 68
  max_mtu: 65535
  flags:
  - broadcast
  - lower_up
  - controller
  - multicast
  - running
  - up
  ipv4:
    addresses:
    - address: 10.211.14.70
      prefix_len: 22
      valid_lft: forever
      preferred_lft: forever
  mac_address: 52:54:00:61:88:fe
  ethtool:
    features:
      fixed:
        fcoe-mtu: false
        hsr-dup-offload: false
        hsr-fwd-offload: false
        hsr-tag-ins-offload: false
        hsr-tag-rm-offload: false
        hw-tc-offload: false
        l2-fwd-offload: false
        loopback: false
        macsec-hw-offload: false
        netns-local: true
        rx-all: false
        rx-checksum: false
        rx-fcs: false
        rx-gro-hw: false
        rx-hashing: false
        rx-ntuple-filter: false
        rx-udp_tunnel-port-offload: false
        tls-hw-record: false
        tls-hw-rx-offload: false
        tls-hw-tx-offload: false
        tx-checksum-fcoe-crc: false
        tx-checksum-ipv4: false
        tx-checksum-ipv6: false
        tx-checksum-sctp: false
        tx-fcoe-segmentation: false
        tx-gso-partial: false
        tx-gso-robust: false
        tx-lockless: true
        tx-tunnel-remcsum-segmentation: false
        tx-vlan-hw-insert: true
        tx-vlan-stag-hw-insert: true
        vlan-challenged: false
      changeable:
        esp-hw-offload: true
        esp-tx-csum-hw-offload: true
        highdma: true
        rx-gro: true
        rx-gro-list: false
        rx-lro: true
        rx-udp-gro-forwarding: false
        rx-vlan-filter: true
        rx-vlan-hw-parse: true
        rx-vlan-stag-filter: true
        rx-vlan-stag-hw-parse: true
        tx-checksum-ip-generic: true
        tx-esp-segmentation: true
        tx-generic-segmentation: false
        tx-gre-csum-segmentation: true
        tx-gre-segmentation: true
        tx-gso-list: false
        tx-ipxip4-segmentation: true
        tx-ipxip6-segmentation: true
        tx-nocache-copy: false
        tx-scatter-gather-fraglist: false
        tx-sctp-segmentation: false
        tx-tcp-ecn-segmentation: false
        tx-tcp-mangleid-segmentation: false
        tx-tcp-segmentation: false
        tx-tcp6-segmentation: false
        tx-udp-segmentation: false
        tx-udp_tnl-csum-segmentation: true
        tx-udp_tnl-segmentation: true
    link_mode:
      auto_negotiate: false
      ours: []
      speed: 0
      duplex: unknown
  bond:
    subordinates:
    - ens3
    - ens4
    mode: active-backup
    miimon: 100
    updelay: 0
    downdelay: 0
    use_carrier: true
    arp_interval: 0
    arp_all_targets: any
    arp_validate: none
    primary_reselect: always
    fail_over_mac: none
    resend_igmp: 1
    num_unsol_na: 1
    num_grat_arp: 1
    all_subordinates_active: dropped
    min_links: 0
    lp_interval: 1
    peer_notif_delay: 0
    arp_missed_max: 2
```

Please let me know if you need any additional data or info.

Thanks for your help.

Venkat

**Comment by cathay4t** [comment]
@VenkatramananV Can you provide a reproduce steps for me to try in Oracle Linux VM?

**Comment by cathay4t** [comment]
The `npc` output cannot explains why nmstate fails. So I have to reproduce this locally to debug.

**Comment by cathay4t** [comment]
This is fixed by https://github.com/rust-netlink/netlink-packet-route/pull/73

Will automatically fix in nmstate once nispor use new version of it.

**Comment by cathay4t** [comment]
This is fixed in 2.2.22 release. Closing
